Assets/Scripts/SceneController.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class SceneController : MonoBehaviour
+{
+    public class SceneState
+    {
+        public string SceneName { get => _sceneName; }
+    
+        private string _sceneName;
+        public SceneState(string sceneName)
+        {
+            _sceneName = sceneName;
+        }
+    }
+    public static SceneController Instance { get; private set; }
+    public SceneState MainMenu { get => _mainMenu; }
+    public SceneState JoinMenu { get => _joinMenu; }
+    public SceneState Game { get => _game; }
+    
+    private SceneState _mainMenu;
+    private SceneState _joinMenu;
+    private SceneState _game;
+    private void Awake()
+    {
+        Instance = this;
+        _game = new("Game");
```

Comment:
Bedre om du skriver `new SceneState("Game")` i stedet for. Man skal skjønne hva slags type en variabel/felt har i konteksten til metoden den er i.